Shared/Samples/Choose camera controller/ChooseCameraControllerView.swift

```diff
@@ -0,0 +1,161 @@
+                    spatialReference: .wgs84
+                ),
+                distance: 5500,
+                heading: 150,
+                pitch: 20,
+                roll: 0
+            )
+        )
+        
+        // Creates a surface.
+        let surface = Surface()
+        
+        // Creates a tiled elevation source.
+        let worldElevationServiceURL = URL(
+            string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"
+        )!
+        let elevationSource = ArcGISTiledElevationSource(url: worldElevationServiceURL)
+        
+        // Adds the elevation source to the surface.
+        surface.addElevationSource(elevationSource)
+        
+        // Sets the surface to the scene's base surface.
+        scene.baseSurface = surface
+        return scene
+    }()
+    
+    /// A graphics overlay containing an airplane graphic.
+    private static let graphicsOverlay: GraphicsOverlay = {
+        let graphicsOverlay = GraphicsOverlay(graphics: [Graphic.airplane])
+        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
+        return graphicsOverlay
+    }()
+    
+    /// A Boolean value indicating whether the settings view should be presented.
+    @State var isShowingSettings = false
+    
+    /// The sun lighting mode of the scene view.
+    @State private var cameraControllerKind: CameraControllerKind = .globe
+    
+    enum CameraControllerKind: CaseIterable {
+        case globe
+        case plane
+        case crater
+        
+        /// A human-readable label for the camera controller kind.
+        var label: String {
+            switch self {
+            case .globe: return "Free pan round the globe"
+            case .plane: return "Orbit camera around plane"
+            case .crater: return "Orbit camera around crater"
+            }
+        }
+    }
+    
+    /// The camera controller of the scene view.
+    @State private var cameraController: CameraController {
+        didSet {
+            _ = GlobeCameraController()
+        }
+    }
+    
+    init() {
+        cameraController = Self.makeCameraController(kind: .crater)
+    }
+    
+    var body: some View {
+        VStack {
+            SceneView(
+                scene: scene,
+                cameraController: cameraController,
+                graphicsOverlays: [ChooseCameraControllerView.graphicsOverlay]
+            )
+            Menu("Camera Controllers") {
+                ForEach(CameraControllerKind.allCases, id: \.self) { kind in
+                    Button(kind.label) {
+                        cameraControllerKind = kind
+                    }
+                }
+            }
+            .onChange(of: cameraControllerKind) { newValue in
+                cameraController = Self.makeCameraController(kind: newValue)
+            }
+        }
+    }
+    
+    static func makeCameraController(kind: CameraControllerKind) -> CameraController {
```

Comment:
Instead of having this method as static, I'd suggest the following changes:

- Make it an instance method
- Call the method to set the `cameraController` on the view in the button, instead of using `onChange` modifier